server/src/controllers/course.ts

```diff
@@ -40,3 +86,60 @@ export async function addCourse(req: Request, res: Response): Promise<void> {
     });
   }
 }
+
+export async function fetchAllInstancesFromSisu(req: Request, res: Response): Promise<Response> {
+  try {
+    const courseId: string = String(req.params.courseId);
+    const instancesFromSisu: AxiosResponse = await axios.get(`${SISU_API_URL}/courseunitrealisations`, {
+      timeout: axiosTimeout,
+      params: {
+        code: courseId,
+        USER_KEY: SISU_API_KEY
+      }
+    });
+
+    if (instancesFromSisu.data?.error) throw new Error(instancesFromSisu.data.error.message);
+    const parsedInstances: Array<InstanceData> = instancesFromSisu.data.map((instance: SisuInstance) => parseSisuInstance(instance));
+
+    return res.status(200).send({
+      success: true,
+      instances: parsedInstances
+    });
+  } catch (error: unknown) {
+    console.log(error);
+
+    return res.status(500).send({
```

Comment:
Should the status code / response body be more specific about what went wrong? E. g. 404 if the course was not found.